### PR TITLE
Publish developer docs

### PR DESCRIPTION
**Pull request by fisx**
https://wearezeta.atlassian.net/browse/SQSERVICES-1724

## Checklist

 - [x] Make sure you've incorporated all changes introduced between a0efd6a0e61c56e7283ce389c6d720dbd0888cdf and develop.
 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

**Comment by fisx** [comment]
ping @smatting

**Comment by fisx** [comment]
- still struggling with rst / md syntax and semantics
- not sure if i want to keep `/docs/{developer,reference,legacy}/...` as links or text references to the new urls.  if we keep them, they will likely deteriorate (see next point), but they may still point roughly into the right direction.
- i am pretty confident i don't want to change the structure or contents of what i've moved any further; that should be work for future PRs.

**Comment by arthurwolf** [comment]
note: go through all references in index/toc and turn # into ##, should fix some warnings

File: docs/src/developer/index.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+re-ordering, and they are far from complete, but we hope they will still
+help you.
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Contents:
+   :glob:
+
+   developer/api-versioning.md <./developer/api-versioning.md>
+   developer/assandra-interaction.md <./developer/cassandra-interaction.md>
```
Comment (yu-pri) [inline docs/src/developer/index.rst:15]:
```suggestion
   developer/cassandra-interaction.md <./developer/cassandra-interaction.md>
```

Comment (fisx) [reply docs/src/developer/index.rst]: thanks!

**Comment by fisx** [comment]
@arthurwolf if you want to resolve the conflicts, you may want to look at what i've tried to do and just start over from the new tip of develop, to make sure you're not missing anything from the old places.

use this PR and force-push something that works, or use a new branch and PR and close this one, whatever's easier for you.

(or you can just give me the proof of concept here and i'll try to find the time to clean it up later.)

**Comment by fisx** [comment]
I would merge this as-is since it's a clear improvement, but here is another thing about rst that gets me mad:

2022-08-24-145428_1920x1080_scrot

On https://docs.wire.com/developer/developer/index.html (or `/docs/src/developer/index.rst`), some of the links in the TOC are labelled with the title of the sub-page, some with the file name (which I am explicitly giving in the TOC, and would expect to be used).  I can't find the difference in the code that triggers this difference in behavior.

I'll keep staring at it for a little while longer, but @arthurwolf maybe you have an idea?  But more importantly please see if you think it can be merged.  I want to wrap this up.

Thanks for all your help with this!

**Comment by smatting** [comment]
The TOC seems to get broken if there are multiple top-level headings in the markdown files. I fixed the heading levels in all the markdown files and added the `developer/` `reference/` structure with this result:

image

Also I removed all tombstone links.

**Review by smatting — APPROVED** [review]

**Pull request merged**